OneCommunityGlobal/HighestGoodNetworkApp
# Imran_See_User_Management_Tab_Full_Functionality

# Description
Create "See User Management Tab (Full Functionality)" permission.
Make the "Other Links" -> "User Management" option appear/accessible with full functionality of User Management page (user is able to create, delete, and update users) for users who have the "See User Management Tab (Full Functionality)" permission. 

## Related PRS (if any):
This frontend PR is related to the [#1039](https://github.com/OneCommunityGlobal/HGNRest/pull/1039) PR.

## Main changes explained:
- add new permission 'userManagementFullFunctionality'
- Updated UserTableData.jsx with new Permission
- Update UserTableData.test.js

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Log in as Owner -> Other LInks -> Permissions Management -> Manage User Permissions
5. Enter the username of a user that does not inherently have the “See User Management Tab (Full Functionality)” permission (i.e., a Volunteer). Click on Submit.
6. Login as the chosen user (if already logged in, make sure to log out and log back in). The “Other Links” button should appear with the “User Management” option available with access to the full functionality of the page (create, delete, and update users).

## Screenshots or videos of changes:
https://github.com/user-attachments/assets/f42ef2af-738c-42fa-b2cb-9b2bbcfba11f


## Note:
Related bug: See below the image of a popup message displayed when a user with 'User Management' permission attempts to delete another user. The popup lacks a button to finalize the deletion process.
![bug](https://github.com/user-attachments/assets/56cef313-7a18-443e-9e2c-1ba07927fea4)
